src/game_main.cpp

```diff
@@ -408,7 +409,18 @@ void initAndRunGame(
 ) {
   auto run = [&](const CommandLineOptions& options) {
     Game game(options, &userProfile, pWindow);
-    return game.run();
+    
+    
+    std::vector<SDL_Event> eventQueue;
+    game.mLastTime = high_resolution_clock::now();
+    for(;;){
+      bool breakOut = false;
+      game.runOneFrame(eventQueue, breakOut);
+      if (breakOut){
+        break;
+      }
+    }
+    return Game::RunResult::GameEnded; //TODO : figure out restart case
```

Comment:
To handle the restart case, my suggestion would be to:

* Add `RunResult::StillRunning` to the `RunResult` enum
* Make `Game::runOneFrame` have a `RunResult` return value

Then, this code here would become:

```
for (;;) {
  const auto result = game.runOneFrame(eventQueue);
  if (result != RunResult::Still running) {
    return result;
  }
}
```

Reply:
yup, can do that. I will make this change meanwhile you can review rest of the parts of the PR

Reply:
I've done that now in https://github.com/lethal-guitar/RigelEngine/pull/561.